azurerm/resource_arm_function_app.go

```diff
@@ -153,6 +155,26 @@ func resourceArmFunctionApp() *schema.Resource {
 							Type:     schema.TypeString,
 							Computed: true,
 						},
+						"user_assigned_identity": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Required: true,
```

Comment:
since this needs to be a UUID could we validate this via:

```suggestion
										Required:     true,
										ValidateFunc: validate.UUID,
```

Reply:
Thanks for the review @tombuildsstuff!
I've fixed everything except this one as it's actually not a UUID but a Azure resource ID as defined [here](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.resourcemanager.models.identity.userassignedidentities?view=azure-dotnet): 

`'/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}'.`

Should we validate it in any other way? I think that could be hard to do as Microsoft could potentially change the format in the future?

Reply:
good spot - we can make this:

```suggestion
										Required: true,
										ValidateFunc: azure.ValidateResourceID,
```

which will validate that it's an Azure Resource ID. For some resources we've started making more granular Resource ID validation functions; however since one of these doesn't exist for UserAssignedIdentities at this time we should be good to just validate it's a Resource ID for the moment 👍

Reply:
@tombuildsstuff Awesome, thanks again! I've updated the PR 🎊